src/pointerAction.js

```diff
@@ -383,8 +383,8 @@ export function dndzone(node, options) {
         const currentIdx = elToIdx.get(originalDragTarget);
         originIndex = currentIdx;
         originDropZone = originalDragTarget.parentElement;
-        /** @type {ShadowRoot | HTMLDocument} */
-        const rootNode = originDropZone.getRootNode();
+        /** @type {ShadowRoot | HTMLDocument | Element } */
+        const rootNode = originDropZone.closest("dialog") || originDropZone.getRootNode();
```

Comment:
i wonder if it is possible to have a shadow root inside of a dialog. if it still works if you reverse the order of the || condition that might be safer

Reply:
I could check and change the implementation, or I could follow the strategy and the limitations you mentioned here instead: https://github.com/isaacHagoel/svelte-dnd-action/issues/455, do you have a preference?

Reply:
I think your idea is simpler and less open to abuse then allowing the user to specify the root element (i hope I understood your question correctly).
I just asked the question to make sure there are no weird edge cases we are not thinking of.
If u can test with shadow root inside of a dialog it would be great (no need to reverse the || just check what happens)

Reply:
I will check after regular work and get back to you 👍

Reply:
Hi any updates on this one?

Reply:
Hi Isaac! Sorry, this got forgotten in the daily churn of stuff. Thanks for pinging me. I've confirmed that a `.closest()` call does not propagate through a shadow DOM root (as it is treated as a document root) in latest chrome and firefox. Does that resolve your concerns enough?

Reply:
yes. can you please pull from master, bump the version in package.json and update release-notes.md.
once you do i can merge

Reply:
Here you go!